app/controllers/CyaCreateRecordController.scala

```diff
@@ -63,9 +65,14 @@ class CyaCreateRecordController @Inject() (
   }
 
   def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
-    auditService
-      .auditFinishCreateGoodsRecord(request.eori, request.affinityGroup, request.userAnswers)
-      .map(_ => Redirect(routes.CreateRecordSuccessController.onPageLoad().url))
+    CreateGoodsRecord.build(request.userAnswers, request.eori) match {
+      case Right(model) =>
+        for {
+          goodsRecordResponse <- goodsRecordConnector.submitGoodsRecord(model)
+          _                   <- auditService.auditFinishCreateGoodsRecord(request.eori, request.affinityGroup, request.userAnswers)
```

Comment:
this needs to be a <- because it is designed to fail if the audit doesn't work (see other controllers)

Reply:
(To be extra pedantic, the audit itself can fail (e.g. the auditservice can give us back an error response), it's only if the Future fails - e.g. connection issues etc. we'd end up failing here)

Reply:
should be an = rather than <- for audit as Chris mentioned, the audit is fire and forget

Reply:
done

Reply:
just removed the test that was thinking it should fail